lib/repl.js

```diff
@@ -949,7 +951,32 @@ function REPLServer(prompt,
           self.displayPrompt();
           return;
         }
+      }
+
+      // In the next two if blocks, we do not use os.EOL instead of '\n'
+      // because on Windows it is '\r\n'
+      if (cmd && StringPrototypeIndexOf(cmd, '\n') !== -1) { // If you are editing a multiline command
+        self.history[0] = StringPrototypeReplaceAll(cmd, '\n', '\r');
+      } else if (self[kBufferedCommandSymbol]) { // If a new multiline command was entered
+        // Remove the first N lines from the self.history array
+        // where N is the number of lines in the buffered command
+
+        const lines = StringPrototypeSplit(self[kBufferedCommandSymbol], '\n');
+        self.history = ArrayPrototypeSlice(self.history, lines.length);
+        ArrayPrototypePop(lines);
+        // And replace them with the single command split by '\r'
+        ArrayPrototypePush(lines, cmd);
```

Comment:
I am not certain why the last entry is replaced here. Could that be added as comment?

```suggestion
        lines[lines.length - 1] = cmd;
```